keps/sig-node/0034-20190130-node-os-arch-labels.md

```diff
@@ -0,0 +1,92 @@
+---
+kep-number: 34
+title: 
+authors:
+  - "@yujuhong"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+reviewers:
+  - "@liggitt"
+approvers:
+  - "@dchen1107"
+editor: Yu-Ju Hong
+creation-date: 2019-01-30
+last-updated: 2019-01-30
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Promote Node Operating System & Archtecture labels to GA
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+
+## Summary
+
+The kubelet has been been labeling the Node object with operating system (OS)
+and archietcture (arch) labels since Kubernetes 1.3. This proposal aims to
+promote these labels to GA and ensure a smooth transition with backward
+compatibility.
+
+Below lists the labels to promote:
+```
+beta.kubernetes.io/os
+beta.kubernetes.io/arch
+```
+
+## Motivation
+
+The labels have existed over two years and are widely in various places.
+Promoting these labels to reflect their stable status.
+
+### Goals
+
+Promote the OS and arch labels to GA.
+
+### Non-Goals
+
+Promoting any other label that is not listed in the Goals sections.
+
+## Proposal
+
+A new set of GA labels will be added:
+
+```
+kubernetes.io/os
+kubernetes.io/arch
+```
+
+Starting in *v1.14*,
+- kubelets will report *both* the beta and the GA labels. 
+- the node controller will reconcile a mismatch between a present or missing
+  GA label in favor of the beta label
+  * This ensure uniform labeling in the presence of pre-1.14 kubelets that
+    don't report the GA labels.
+
+Starting in *v1.18*, 
+- kubelet will stop reporting the beta labels.
+- the node controller will switch to reconciling a mismatch between a present
```

Comment:
does the node controller do this forever?  i am wondering if there is a point in time where users must update their manifests at all?

Reply:
The "present" wording was intentional, but probably too subtle. This would only modify an existing beta label to match the GA label, it would not add it, so once the kubelet stopped recording the beta label (1.18), newly registered nodes would not have the label. Call that out explicitly in the doc.

Reply:
I expanded both the 1.14 and 1.18 paragraph to make it more clear what the node controller would do. PTAL

Reply:
so users eventually are required to update node selectors in their manifests to the new label between 1.14 and 1.18 in order to steer their workloads appropriately.  that is good to understand and i am fine with that.  it could impact some folks that did a backup/restore via something like ark across cluster versions, but i think this is sensible enough.

Reply:
Yes, 1.18 is chosen based on the existing deprecation policy.